Client/UserResearch/OnboardingUserResearch.swift

```diff
@@ -4,25 +4,80 @@
 
 import Foundation
 import Leanplum
+import Shared
 
 struct LPVariables {
     static var showOnboardingScreen = LPVar.define("showOnboardingScreen", with: true)
 }
 
+enum OnboardingScreenType: String {
+    case versionV1 // Default
+    case versionV2 // New version 2
+}
+
 class OnboardingUserResearch {
-    // Delegate closure
+    // Closure delegate
     var updatedLPVariables: ((LPVar?) -> Void)?
     // variable
     var lpVariable: LPVar?
+    // Constants
+    private let onboardingScreenTypeKey = "onboardingScreenTypeKey"
+    // Saving user defaults
+    private let defaults = UserDefaults.standard
+    // Publicly accessible onboarding screen type
+    var onboardingScreenType:OnboardingScreenType? {
+        set(value) {
+            if value == nil {
+                defaults.removeObject(forKey: onboardingScreenTypeKey)
+            } else {
+                defaults.set(value?.rawValue, forKey: onboardingScreenTypeKey)
+            }
+        }
+        get {
+            guard let value = defaults.value(forKey: onboardingScreenTypeKey) else {
+                return nil
+            }
+            return OnboardingScreenType(rawValue: value as! String) ?? nil
+        }
+    }
     
-    // Initializer
+    // MARK: Initializer
     init(lpVariable: LPVar? = LPVariables.showOnboardingScreen) {
         self.lpVariable = lpVariable
     }
     
+    // MARK: public
     func lpVariableObserver() {
         Leanplum.onVariablesChanged {
             self.updatedLPVariables?(self.lpVariable)
         }
     }
+    
+    func updateValue(value: Bool) {
+        // For LP variable below is the convention
+        // we are going to follow
+        // True = Current Onboarding Screen
+        // False = New Onboarding Screen
+        onboardingScreenType = value ? .versionV1 : .versionV2
+    }
+    
+    func updateTelemetry() {
+        print("lp variant \(String(describing: Leanplum.variants()))")
+        if Leanplum.variants() != nil {
+            let lpVariants = Leanplum.variants()?.first
+            if let lpData = lpVariants as? Dictionary<String, AnyObject> {
+                var abTestId = ""
+                if let value = lpData["abTestId"] as? Int64 {
+                    abTestId = "\(value)"
+                }
+                let abTestName = lpData["abTestName"] as? String ?? ""
+                let abTestVariant = lpData["name"] as? String ?? ""
+                let attributesExtras = [LPAttributeKey.experimentId: abTestId, LPAttributeKey.experimentName: abTestName, LPAttributeKey.experimentVariant: abTestVariant]
+                // Leanplum telemetry
+                LeanPlumClient.shared.set(attributes: attributesExtras)
```

Comment:
**Note:** 
**Leanplum:** The attributes in LP are per session. Unlike variables these don't accumulate but change as soon as we update the event. Idea is to send them once per session or when user's condition changes for the experiment. 
**Unified Telemetry:** Currently we don't have a way to send attributes hence sending the same information to unified telemetry via new category and event type of enrollment. Hoping this changes with Glean